bdio2/src/main/java/com/blackducksoftware/bdio2/BdioObject.java

```diff
@@ -130,6 +130,16 @@ public final String id() {
         return ExtraStrings.beforeLast((String) value, '#');
     }
 
+    /**
+     * Returns the scanType for this object.
+     */
+    @Nullable
```

Comment:
Same as above.

Reply:
Here `scanType` needs to be nullable , similar to what's done to `id`, so that while reading the document correct scan type can be added to the new document.